Sources/Overlays/_Testing_CoreGraphics/Attachments/Attachment+AttachableAsCGImage.swift

```diff
@@ -0,0 +1,147 @@
+    // `preferredContentType(forEncodingQuality:)` but will go away once our
+    // minimum deployment targets include the UniformTypeIdentifiers framework.)
+    var preferredName = preferredName ?? Self.defaultPreferredName
+    if #available(_uttypesAPI, *) {
+      let contentType: UTType = contentType
+        .map { $0 as! UTType }
+        .flatMap { contentType in
+          if UTType.image.conforms(to: contentType) {
+            // This type is an abstract base type of .image (or .image itself.)
+            // We'll infer the concrete type based on other arguments.
+            return nil
+          }
+          return contentType
+        } ?? .preferred(forEncodingQuality: encodingQuality)
+      preferredName = (preferredName as NSString).appendingPathExtension(for: contentType)
+      imageContainer.contentType = contentType
+    } else {
+      // The caller can't provide a content type, so we'll pick one for them.
+      let ext = if encodingQuality < 1.0 {
+        "jpg"
+      } else {
+        "png"
+      }
+      if (preferredName as NSString).pathExtension.caseInsensitiveCompare(ext) != .orderedSame {
+        preferredName = (preferredName as NSString).appendingPathExtension(ext) ?? preferredName
+      }
+    }
+
+    self.init(imageContainer, named: preferredName, sourceLocation: sourceLocation)
+  }
+
+  /// Initialize an instance of this type that encloses the given image.
+  ///
+  /// - Parameters:
+  ///   - attachableValue: The value that will be attached to the output of
+  ///     the test run.
+  ///   - preferredName: The preferred name of the attachment when writing it
+  ///     to a test report or to disk. If `nil`, the testing library attempts
+  ///     to derive a reasonable filename for the attached value.
+  ///   - contentType: The image format with which to encode `attachableValue`.
+  ///     If this type does not conform to [`UTType.image`](https://developer.apple.com/documentation/uniformtypeidentifiers/uttype-swift.struct/image),
+  ///     the result is undefined. Pass `nil` to let the testing library decide
+  ///     which image format to use.
+  ///   - encodingQuality: The encoding quality to use when encoding the image.
+  ///     If the image format used for encoding (specified by the `contentType`
+  ///     argument) does not support variable-quality encoding, the value of
+  ///     this argument is ignored.
+  ///   - sourceLocation: The source location of the call to this initializer.
+  ///     This value is used when recording issues associated with the
+  ///     attachment.
+  ///
+  /// The following system-provided image types conform to the
+  /// ``AttachableAsCGImage`` protocol and can be attached to a test:
+  ///
+  /// - [`CGImage`](https://developer.apple.com/documentation/coregraphics/cgimage)
+  @_spi(Experimental)
+  @available(_uttypesAPI, *)
+  public init<T>(
+    _ attachableValue: T,
+    named preferredName: String? = nil,
+    as contentType: UTType?,
```

Comment:
```suggestion
    contentType: UTType?,
```

Reply:
No, that one's on purpose so that it reads when used:

```swift
Attachment(something, as: .png)
```

Which is consistent with other uses of `UTType` in Apple API.

Reply:
It's just that it can be unclear when it's `as: nil`. Do the other Apple APIs accept Optional `UTType` values?

Reply:
Yes, but also there's an overload that doesn't take a content type at all, and I'd expect people to use that instead of a literal `nil`. The optionality here is so that you can pass in an optional value rather than needing to manually branch with `if`.